packages/twenty-server/src/engine/core-modules/billing/entities/billing-product.entity.ts

```diff
@@ -0,0 +1,55 @@
+import { Field, registerEnumType } from '@nestjs/graphql';
+
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  OneToMany,
+  PrimaryGeneratedColumn,
+  Relation,
+  UpdateDateColumn,
+} from 'typeorm';
+
+import { BillingPrice } from 'src/engine/core-modules/billing/entities/billing-price.entity';
+import { BillingAvailablePlanKey } from 'src/engine/core-modules/billing/enums/billing-available-plan-key.enum';
+import { BillingUsageType } from 'src/engine/core-modules/billing/enums/billing-usage-type.enum';
+
+registerEnumType(BillingUsageType, { name: 'BillingUsageType' });
+
+@Entity({ name: 'billingProduct', schema: 'core' })
+export class BillingProduct {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ nullable: true, type: 'timestamptz' })
+  deletedAt?: Date;
+
+  @CreateDateColumn({ type: 'timestamptz' })
+  createdAt: Date;
+
+  @UpdateDateColumn({ type: 'timestamptz' })
+  updatedAt: Date;
+
+  @Column({ nullable: false })
+  active: boolean;
+
+  @Column({ nullable: false })
+  stripeProductId: string;
+
+  @Column({ nullable: false })
+  defaultPriceId: string;
+
+  @Column({ nullable: false })
+  defaultStripePriceId: string;
+
+  @Field(() => BillingAvailablePlanKey, { nullable: false })
+  @Column({
+    type: 'enum',
+    enum: Object.values(BillingAvailablePlanKey),
+    nullable: false,
+  })
+  planKey: BillingAvailablePlanKey;
```

Comment:
Maybe you've discussed that with @charlesBochet  but I guess this comes from metadata column? I would be tempted to use a jsonb metadata column and then cast at the code level to really stick to Stripe as much as we can

Reply:
agree, + let's keep some flexibility here as this is flexible in stripe